tools/yul_phaser/Chromosome.h

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace solidity::phaser
+{
+
+class Chromosome
+{
+public:
+	Chromosome():
+		m_optimizationSteps{} {}
+	explicit Chromosome(std::vector<std::string> const& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	explicit Chromosome(std::vector<std::string>&& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	static Chromosome make_random(int _length);
+
+	size_t length() const { return m_optimizationSteps.size(); }
+
+	std::vector<std::string> const& asSequence() const { return m_optimizationSteps; }
+	friend std::ostream& operator<<(std::ostream& _stream, Chromosome const& _chromosome);
+
+private:
+	static char abbreviateStepName(std::string const& _stepName);
```

Comment:
Can you pull these out somewhere else (into the same directory where the OptimizerSuite is)? They are too generic for the Chromosome, I would say. Maybe the OptimiserSuite can even use the pulled out function then.

Reply:
Sure. I'll see what I can do. In that cause I should probably make `yulopti` use that too.

Reply:
sounds good! yulopti could even auto-generate the help screen from the names and abbreviations at some point...

Reply:
Done.

I have added a pair of maps between `char` and `string` to `OptimiserSuite`. Maps are cached in static variables (just like in `allSteps()` and one is generated from the other so they won't go out of sync. I considered using `boost::bimap` or `InvertibleMap` from `libsolutil` but both seem like an overkill for a small bidirectional map that never changes.

`yulopti` has been updated, including the help screen which is now autogenerated and looks like this:
```
p: StackCompresor                q: quit                          C: ConditionalSimplifier         D: DeadCodeEliminator            
I: ForLoopConditionIntoBody      L: LoadResolver                  M: LoopInvariantCodeMotion       O: ForLoopConditionOutOfBody     
T: LiteralRematerialiser         U: ConditionalUnsimplifier       V: SSAReverser                   a: SSATransform                  
c: CommonSubexpressionEliminator d: VarDeclInitializer            e: ExpressionInliner             f: BlockFlattener                
g: FunctionGrouper               h: FunctionHoister               i: FullInliner                   j: ExpressionJoiner              
l: VarNameCleaner                m: Rematerialiser                n: ControlFlowSimplifier         o: ForLoopInitRewriter           
r: RedundantAssignEliminator     s: ExpressionSimplifier          t: StructuralSimplifier          u: UnusedPruner                  
v: EquivalentFunctionCombiner    x: ExpressionSplitter            ?
```

All the changes external to `yulPhaser` are in three commits at the bottom of the branch. I can pull them out into a separate PR if needs be.

Reply:
This code is now in #8215.